Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/Ride.hs

```diff
@@ -151,16 +152,17 @@ data StuckRideItem = StuckRideItem
     riderId :: Id Person
   }
 
-findStuckRideItems :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Id Merchant -> [Id Booking] -> UTCTime -> m [StuckRideItem]
-findStuckRideItems (Id merchantId) bookingIds now = do
+findStuckRideItems :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Id Merchant -> Id DMOC.MerchantOperatingCity -> [Id Booking] -> UTCTime -> m [StuckRideItem]
+findStuckRideItems (Id merchantId) moCityId bookingIds now = do
   let now6HrBefore = addUTCTime (- (6 * 60 * 60) :: NominalDiffTime) now
-  bookings <-
+  result <-
     findAllWithDb
       [ Se.And
           [ Se.Is BeamB.providerId $ Se.Eq merchantId,
             Se.Is BeamB.id $ Se.In $ getId <$> bookingIds
           ]
       ]
+  let bookings = filter (\x -> moCityId == x.merchantOperatingCityId) result
```

Comment:
Why not inside the query only. Apply same in all other function changes.

Reply:
Done